tests/columns/test_m2m.py

```diff
@@ -0,0 +1,328 @@
+            ],
+        )
+
+        # Now try it in reverse.
+        response = Genre.select(Genre.name, Genre.bands(Band.name)).run_sync()
+        self.assertEqual(
+            response,
+            [
+                {"name": "Rock", "bands": ["Pythonistas", "C-Sharps"]},
+                {"name": "Folk", "bands": ["Pythonistas", "Rustaceans"]},
+                {"name": "Classical", "bands": ["C-Sharps"]},
+            ],
+        )
+
+    def test_select_id(self):
+        response = Band.select(Band.name, Band.genres(Genre.id)).run_sync()
+        self.assertEqual(
+            response,
+            [
+                {"name": "Pythonistas", "genres": [1, 2]},
+                {"name": "Rustaceans", "genres": [2]},
+                {"name": "C-Sharps", "genres": [1, 3]},
+            ],
+        )
+
+        # Now try it in reverse.
+        response = Genre.select(Genre.name, Genre.bands(Band.id)).run_sync()
+        self.assertEqual(
+            response,
+            [
+                {"name": "Rock", "bands": [1, 3]},
+                {"name": "Folk", "bands": [1, 2]},
+                {"name": "Classical", "bands": [3]},
+            ],
+        )
+
+    def test_add_m2m(self):
+        """
+        Make sure we can add items to the joining table.
+        """
+        band: Band = Band.objects().get(Band.name == "Pythonistas").run_sync()
+        band.add_m2m(Genre(name="Punk Rock"), m2m=Band.genres).run_sync()
+
+        self.assertTrue(
+            Genre.exists().where(Genre.name == "Punk Rock").run_sync()
+        )
+
+        self.assertEqual(
+            GenreToBand.count()
+            .where(
+                GenreToBand.band.name == "Pythonistas",
+                GenreToBand.genre.name == "Punk Rock",
+            )
+            .run_sync(),
+            1,
+        )
+
+    def test_add_m2m_existing(self):
+        """
+        Make sure we can add an existing element to the joining table.
+        """
+        band: Band = Band.objects().get(Band.name == "Pythonistas").run_sync()
+
+        genre: Genre = (
+            Genre.objects().get(Genre.name == "Classical").run_sync()
+        )
+
+        band.add_m2m(genre, m2m=Band.genres).run_sync()
+
+        # We shouldn't have created a duplicate genre in the database.
+        self.assertEqual(
+            Genre.count().where(Genre.name == "Classical").run_sync(), 1
+        )
+
+        self.assertEqual(
+            GenreToBand.count()
+            .where(
+                GenreToBand.band.name == "Pythonistas",
+                GenreToBand.genre.name == "Classical",
+            )
+            .run_sync(),
+            1,
+        )
+
+    def test_get_m2m(self):
+        """
+        Make sure we can get related items via the joining table.
+        """
+        band: Band = Band.objects().get(Band.name == "Pythonistas").run_sync()
+
+        genres = band.get_m2m(Band.genres).run_sync()
+
+        self.assertTrue(all([isinstance(i, Table) for i in genres]))
```

Comment:
```suggestion
        self.assertTrue(all(isinstance(i, Table) for i in genres))
```

Functions like `any`, `all` and `sum` allow you to pass in a generator rather than a collection. Doing so removes a pair of brackets, making the intent slightly clearer. It will also return immediately if a hat is found, rather than having to build the whole list.